pallets/runtime/src/bridge.rs

```diff
@@ -0,0 +1,303 @@
+        ChangeValidatorsProposals get(change_validators_proposals): map T::AccountId => Option<u64>;
+        /// Correspondence between bridge transaction proposals and multisig proposal IDs.
+        BridgeTxProposals get(bridge_tx_proposals): map hasher(blake2_256) BridgeTx<T::AccountId> => Option<u64>;
+        /// Pending issuance transactions to identities.
+        PendingTxs get(pending_txs): map IdentityId => Vec<BridgeTx<T::AccountId>>;
+    }
+}
+
+/// The intended recipient of POLY exchanged from the locked ERC20 tokens.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IssueRecipient<AccountId> {
+    Account(AccountId),
+    Identity(IdentityId),
+}
+
+/// A unique lock-and-mint bridge transaction containing Ethereum transaction data and a bridge nonce.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BridgeTx<AccountId> {
+    /// Bridge validator runtime nonce.
+    pub nonce: u64,
+    /// Recipient of POLY on Polymesh: the deposit address or identity.
+    pub recipient: IssueRecipient<AccountId>,
+    /// Amount of tokens locked on Ethereum.
+    pub value: u128,
+    /// Ethereum token lock transaction hash.
+    pub tx_hash: H256,
+}
+
+/// A transaction that is pending a valid identity KYC.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PendingTx<AccountId> {
+    /// The identity on which the KYC is pending.
+    pub did: IdentityId,
+    /// The pending transaction.
+    pub bridge_tx: BridgeTx<AccountId>,
+}
+
+/// Either a pending transaction or a `None` or an error.
+type IssueResult<AccountId> = sp_std::result::Result<Option<PendingTx<AccountId>>, DispatchError>;
+
+decl_event! {
+    pub enum Event<T> where AccountId = <T as frame_system::Trait>::AccountId {
+        /// Confirmation of a validator set change.
+        ValidatorsChanged(AccountId),
+        /// Confirmation of minting POLY on Polymesh in return for the locked ERC20 tokens on
+        /// Ethereum.
+        Bridged(BridgeTx<AccountId>),
+        /// Notification of an approved transaction having moved to a pending state due to the
+        /// recipient identity either being non-existent or not having a valid KYC.
+        Pending(PendingTx<AccountId>),
+        /// Notification of a failure to finalize a pending transaction. The transaction is removed.
+        Failed(BridgeTx<AccountId>),
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Proposes to change the address of the bridge validator multisig accountn, which amounts
+        /// to making a multisig proposal for the validator set change if the change is new or
+        /// approving an existing proposal if the change has already been proposed.
+        pub fn propose_change_validators(origin, new_validators: T::AccountId) -> DispatchResult {
```

Comment:
We don't necessarily need this function. Bridge validators can directly use the multisig module.

Reply:
Possibly, although I'm not sure about the type conversion from `bridge::Call` to `identity::Trait::Proposal` that the bridge validators would need to perform in the multisig module in that case.

Reply:
Validators will be directly using the multisig module to create a proposal to change validators in the validator multisig. It's already possible to create such a proposal from the UI.

Reply:
I'm trying to see if we can do that in the code as well. My unit tests check the ability to change validators and at the moment use `propose_change_validators`.

Reply:
Rather than changing validators through a Proposal/Call, can't we just directly call the `add_multisig_signer` and `remove_multisig_signer` function of the multisig module with validator's origin?

We can also directly manipulate the MultiSigSigners mapping of the multisig module.

Reply:
Can we add a function to replace the signers completely in the multisig module. I think that might be equivalent to changing the address in the bridge module. With add/remove it's possible to get into strange situations where part of the validators are old and part are new even in cases where governance tells to swap the sets atomically. Or we can end up removing all validators and having the empty set and being either stuck or open to any proposals.

Reply:
Currently, we do not store list of signers in the multisig module so it's not directly possible. However, I think it will be a good addition. `map (T::AccountId, Signatory) => bool;` can be converted to `double map hasher(blake2_256) T::AccountId, blake2_256(Signatory) => bool` to allow such enumeration.

That being said, it might be best done outside this PR.